### PR TITLE
[diagnostics] Update dotnet trace docs

### PR DESCRIPTION
**Pull request by mdh1418**
## Summary

Updates `dotnet-trace collect`'s `--output` default value.
Add `dotnet-trace collect`'s `--resume-runtime` option from https://github.com/dotnet/diagnostics/pull/2343.
Add `dotnet-trace collect`'s `--stopping-event-*` options from https://github.com/dotnet/diagnostics/pull/4363.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/diagnostics/dotnet-trace.md](https://github.com/dotnet/docs/blob/63bbd1589c341db95864a2f19b9249d80e63ee0b/docs/core/diagnostics/dotnet-trace.md) | [dotnet-trace performance analysis utility](https://review.learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-trace?branch=pr-en-us-37749) |

### CODE REVIEW

File: docs/core/diagnostics/dotnet-trace.md [hunk 1]
```diff
@@ -200,6 +203,22 @@ dotnet-trace collect [--buffersize <size>] [--clreventlevel <clreventlevel>] [--
...
+
+  A string, parsed as-is, that will stop the trace upon hitting an event with the matching provider name. For a more specific stopping event, additionally provide `--stopping-event-event-name` and/or `--stopping-event-payload-filter`.
+
+- **`--stopping-event-event-name`**
+
+  A string, parsed as-is, that will stop the trace upon hitting an event with the matching event name. Requires `--stopping-event-provider-name` to be set. For a more specific stopping event, additionally provide `--stopping-event-payload-filter`.
+
+- **`--stopping-event-payload-filter`**
+
+  A string, parsed as [payload_field_name]:[payload_field_value] pairs separated by commas, that will stop the trace upon hitting an event with a matching payload. Requires `--stopping-event-provider-name` and `--stopping-event-event-name` to be set.
```
Comment (mikelle-rogers) [inline docs/core/diagnostics/dotnet-trace.md:220]: If there are more than one name:value pairs, will the trace stop on the whichever one comes first?

Comment (mdh1418) [reply docs/core/diagnostics/dotnet-trace.md:220]: A `TraceEvent`'s payload typically will consist of multiple name/values.
For instance,
```
Hit StoppingEvent with ProviderName: `Microsoft-Windows-DotNETRuntime`  EventName: `Method/JittingStarted` at time: `10/24/2023 4:00:16 PM
Payload count: `8`
0 payloadName `MethodID` payloadValue `-5476376640355487248`
1 payloadName `ModuleID` payloadValue `-5476376641966429840`
2 payloadName `MethodToken` payloadValue `100664394`
3 payloadName `MethodILSize` payloadValue `68`
4 payloadName `MethodNamespace` payloadValue `System.AppContext`
5 payloadName `MethodName` payloadValue `Setup`
6 payloadName `MethodSignature` payloadValue `void(char**,char**,int)`
7 payloadName `ClrInstanceID` payloadValue `9`
```

[The filtering logic in `EventMonitor.cs`](https://github.com/dotnet/diagnostics/blob/bb25d231174007a23933dc1996bb752f8fbdabcd/src/Microsoft.Diagnostics.Monitoring.EventPipe/EventMonitor.cs#L99-L110) will only consider a match if all of the user provided pairs are a match.

If one were to pass `--stopping-event-payload-filter MethodNamespace:System.AppContext,MethodName:GetValue`, it wouldn't match the TraceEvent above.

Comment (mikelle-rogers) [reply docs/core/diagnostics/dotnet-trace.md:220]: Would you please add in an example of using the new arguments? And have it clarify that all user provided pairs are a match?

Comment (mdh1418) [reply docs/core/diagnostics/dotnet-trace.md]: Done!

Comment (mikelle-rogers) [reply docs/core/diagnostics/dotnet-trace.md]: Looks great to me!

**Review by mikelle-rogers — APPROVED** [review]
LGTM

**Comment by mdh1418** [comment]
Oops, it doesn't seem like squash + merge is the default... would this be a problem?

**Comment by gewarren** [comment]
> Oops, it doesn't seem like squash + merge is the default... would this be a problem?

It should be fine.

**Pull request merged**